supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/caching/CacheActor.scala

```diff
@@ -0,0 +1,62 @@
+package cromwell.backend.impl.htcondor.caching
+
+import akka.actor.{Actor, ActorLogging}
+import cromwell.backend.BackendJobExecutionActor.SucceededResponse
+import cromwell.backend.impl.htcondor.caching.CacheActor._
+import cromwell.backend.impl.htcondor.caching.exception.{CachedResultAlreadyExistException, CachedResultNotFoundException}
+import cromwell.backend.impl.htcondor.caching.model.CachedExecutionResult
+
+object CacheActor {
+
+  trait CacheActorCommand
+  case class ReadExecutionResult(hash: String) extends CacheActorCommand
+  case class StoreExecutionResult(hash: String, succeededResponse: SucceededResponse) extends CacheActorCommand
+
+  trait CacheActorResponse
+  case class ExecutionResultFound(succeededResponse: SucceededResponse) extends CacheActorResponse
+  case object ExecutionResultNotFound extends CacheActorResponse
+  case class ExecutionResultStored(hash: String) extends CacheActorResponse
+  case object ExecutionResultAlreadyExist extends CacheActorResponse
+
+}
+
+trait CacheActor extends Actor with ActorLogging {
```

Comment:
One more thought i would like to brainstorm with you, You can divide this actor into two separate actors like `CacheActor` Responsibility is only to read/store irrespective flag is enabled or disabled where as other actor is to toggle between enabled cache actor orElse receive message if cache is disabled. By this way you make sure SR is applicable for cache actor. Thoughts ?